waspc/src/Wasp/Generator/Valid/Validator.hs

```diff
@@ -0,0 +1,76 @@
+module Wasp.Generator.Valid.Validator where
+
+import Data.Bifunctor (first)
+import Data.List (intercalate)
+import Data.List.NonEmpty (NonEmpty)
+import qualified Data.List.NonEmpty as NE
+import Data.Maybe (maybeToList)
+import qualified Validation as V
+import Wasp.Util (indent)
+
+type Validator input result = input -> Validation result
+
+type Validation result =
+  V.Validation (NonEmpty ValidationError) result
+
+data ValidationError = ValidationError
+  { message :: String,
+    fieldPath :: [String],
+    fileName :: Maybe String
+  }
+
+-- | Runs the given validator on the input and returns either validation errors or the result.
+runValidator :: Validator input result -> input -> Either (NonEmpty ValidationError) result
+runValidator validator = V.validationToEither . validator
+
+-- | Executes the given validator on the input and returns a list of validation errors. If there are
+-- no errors, returns an empty list.
+execValidator :: Validator input result -> input -> [ValidationError]
```

Comment:
Hm so this sounds identical to `runValidator`, just that it drops the result.

Name doesn't help as they sound like the same thing.

Do we really need both? I would suggest dropping `execValidator` and just inlining that code if you need it only in a place or two. If you really need, let's rename it so it is clear what it does?